core/src/trezor/wire/__init__.py

```diff
@@ -449,6 +459,10 @@ async def handle_session(
 def find_registered_workflow_handler(
     iface: WireInterface, msg_type: int
 ) -> Optional[Handler]:
+    if msg_type in experimental_types and not experimental_enable:
+        # Message is for experimental feature but these are turned off.
+        return None
+
```

Comment:
In some cases the experimental feature might be triggered not by the message type, but by a certain field being set to a certain value. The mechanism in this PR doesn't allow us to specify something like that. I am wondering, how shall we deal with those cases? I think it makes most sense to handle that inside the feature code by looking at `storage.device.get_experimental_features()` or `wire.experimental_enable`, rather than devising some sophisticated mechanism here. Opinions?

Reply:
Agreed, but also I'd strongly prefer not doing that at all.